File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # create a 
        # iterate through array and push to set if empty and if i+1 is in freq map
        # skip if its already part of set

        c = {}

        for i in nums:
            c[i] = c.get(i,0) + 1
        
        count = 0
        ans = 0

        for i in nums:
            if i-1 not in c:
                count = 1
                val = i + 1
                while val in c:
                    val += 1
                    count += 1
            if count > ans:
                ans = count

        return ans 
        

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        
        int l = 0;
        int r = nums.size() - 1;
        int ans = 1000;
        while(l<=r){
            if(nums[l] > nums[r]){
                ans = min(ans,nums[r]);
                r--;
                l++;
            }
            else{
                ans = min(nums[l],ans);
                break;
            }
        }
        return ans;
    }
};
